sklearn/ensemble/_hist_gradient_boosting/binning.py

```diff
@@ -104,24 +120,44 @@ def __init__(self, max_bins=256, subsample=int(2e5), random_state=None):
         self.subsample = subsample
         self.random_state = random_state
 
-    def fit(self, X, y=None):
+    def fit(self, X, support_missing_values=False):
```

Comment:
- > The alternative, where we figure out this from X 

  That's not an alternative. It would be incorrect to look at missing values in `X` because in a GBDT setting the `X` that we pass here is the training data (excluding the validation data). There might be features with missing values in the validation data, that have no missing values in the training data. We need to allocate the first bin if there are missing values in either. I'll try to make this clearer in the docstring.

- keeping the plain `bool` allows to have a default and be able to just call `fit(X)`. It makes the tests less boilerplate. I can remote it.

Reply:
> keeping the plain bool allows to have a default and be able to just call fit(X). It makes the tests less boilerplate

Okay lets keep the `bool` support.